backpack/extensions/secondorder/diag_ggn/__init__.py

```diff
@@ -136,3 +136,67 @@ def __init__(self, mc_samples=1):
 
     def get_num_mc_samples(self):
         return self._mc_samples
+
+
+class BatchDiagGGN(BackpropExtension):
+    VALID_LOSS_HESSIAN_STRATEGIES = [
+        LossHessianStrategy.EXACT,
+        LossHessianStrategy.SAMPLING,
+    ]
+
+    def __init__(self, loss_hessian_strategy=LossHessianStrategy.EXACT, savefield=None):
+        if savefield is None:
+            savefield = "diag_ggn_batch"
+        if loss_hessian_strategy not in self.VALID_LOSS_HESSIAN_STRATEGIES:
+            raise ValueError(
+                "Unknown hessian strategy: {}".format(loss_hessian_strategy)
+                + "Valid strategies: [{}]".format(self.VALID_LOSS_HESSIAN_STRATEGIES)
+            )
```

Comment:
Since this is the parent class, `savefield` and `loss_hessian_strategy` should be set through the child constructor. Can you make both arguments non-keyword? I know this is different in `DiagGGN`, but not having default values for both fields better reflects the role of `BatchDiagGGN` as abstract base class